system/lib/wasmfs/file.h

```diff
@@ -266,33 +289,47 @@ class DataFile::Handle : public File::Handle {
 
 class Directory::Handle : public File::Handle {
   std::shared_ptr<Directory> getDir() { return file->cast<Directory>(); }
+  void finishInsertingChild(const std::string& name,
+                            std::shared_ptr<File> child);
 
 public:
   Handle(std::shared_ptr<File> directory) : File::Handle(directory) {}
   Handle(std::shared_ptr<File> directory, std::defer_lock_t)
     : File::Handle(directory, std::defer_lock) {}
 
-  std::shared_ptr<File> getChild(const std::string& name) {
-    // Unlinked directories must be empty, without even "." or ".."
-    if (!getParent()) {
-      return nullptr;
-    }
-    if (name == ".") {
-      return file;
-    }
-    if (name == "..") {
-      return getParent();
-    }
-    return getDir()->getChild(name);
-  }
+  std::shared_ptr<File> getChild(const std::string& name);
+
+  // Add a child to this directory's entry cache without actually inserting it
+  // in the underlying backend. Assumes a child with this name does not already
+  // exist. Return `true` on success and `false` otherwise.
+  bool mountChild(const std::string& name, std::shared_ptr<File> file);
```

Comment:
(I ask because the comment doesn't say anything about mounting or being a mount, just about not actually inserting it, which is not obviously connected to that - except that a mount has a different backend I guess?)

Reply:
"Mounting" a directory is synonymous with adding that directory to the dcache without affecting the underlying storage, just like on Linux/POSIX. So calling `mountChild` will result in a dcache entry with `DCacheKind::Mount`. I'll add a comment to the enum.

Reply:
This is hopefully clarified by the new comment about the `dcache` that explains mounts.